temporalio/worker/workflow_instance.py

```diff
@@ -0,0 +1,1396 @@
+            async def backoff_then_reschedule() -> None:
+                # Don't pop, we want this to remain as pending
+                handle = self._pending_activities[job.seq]
+                try:
+                    await asyncio.sleep(
+                        job.result.backoff.backoff_duration.ToTimedelta().total_seconds()
+                    )
+                except Exception as err:
+                    self._pending_activities.pop(job.seq)
+                    handle._resolve_failure(err)
+                    return
+                self._pending_commands.append(
+                    handle._build_schedule_command(job.result.backoff)
+                )
+
+            # Schedule it
+            # TODO(cretz): Do we need to do this inside the activity task so it
+            # can be cancelled?
+            self.create_task(backoff_then_reschedule())
+        else:
+            raise RuntimeError("Activity did not have result")
+
+    def _apply_resolve_child_workflow_execution(
+        self,
+        job: temporalio.bridge.proto.workflow_activation.ResolveChildWorkflowExecution,
+    ) -> None:
+        # No matter the result, we know we want to pop
+        handle = self._pending_child_workflows.pop(job.seq, None)
+        if not handle:
+            raise RuntimeError(
+                f"Failed finding child workflow handle for sequence {job.seq}"
+            )
+        if job.result.HasField("completed"):
+            ret: Optional[Any] = None
+            if job.result.completed.HasField("result"):
+                ret_types = [handle._input.ret_type] if handle._input.ret_type else None
+                ret_vals = self._convert_payloads(
+                    [job.result.completed.result],
+                    ret_types,
+                )
+                ret = ret_vals[0]
+            handle._resolve_success(ret)
+        elif job.result.HasField("failed"):
+            handle._resolve_failure(
+                temporalio.exceptions.failure_to_error(
+                    job.result.failed.failure, self._payload_converter
+                )
+            )
+        elif job.result.HasField("cancelled"):
+            handle._resolve_failure(
+                temporalio.exceptions.failure_to_error(
+                    job.result.cancelled.failure, self._payload_converter
+                )
+            )
+        else:
+            raise RuntimeError("Child workflow did not have result")
+
+    def _apply_resolve_child_workflow_execution_start(
+        self,
+        job: temporalio.bridge.proto.workflow_activation.ResolveChildWorkflowExecutionStart,
+    ) -> None:
+        handle = self._pending_child_workflows.get(job.seq)
+        if not handle:
+            raise RuntimeError(
+                f"Failed finding child workflow handle for sequence {job.seq}"
+            )
+        if job.HasField("succeeded"):
+            handle._resolve_start_success(job.succeeded.run_id)
```

Comment:
Could explain that the handle isn't popped here so it can be used for completion

Reply:
While I expected pops vs gets to be obvious, I suppose I can comment saying pop vs get